packages/vanilla/src/types.ts

```diff
@@ -0,0 +1,55 @@
+type AuthResponse = {
```

Comment:
I moved these type definitions to a TS file alongside the source code for a couple reasons. TS will generate definition files (.d.ts files) for any TS file that it compiles, so I prefer not to mix compiled and hand-made .d.ts files if I don't have to. Also, this means the .js and .d.ts files will all wind up in the `./dist` dir, so packaging is a little easier.